### PR TITLE
Feature/cluster autoscaler

### PR DESCRIPTION
**Pull request by maxgio92**
Hi,
it would be nice to allow the node group to autoscale his ASG, besides adding the required tags to it.
In this PR I added a boolean-based flag to "enable" the cluster autoscaler in order to setting the correct tags and adding the needed policy to the NG's Role.

Thank you

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -38,6 +38,34 @@ data "aws_iam_policy_document" "assume_role" {
...
+
+    resources = [
+      "*"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "amazon_eks_worker_node_autoscaler_policy" {
+  count  = (var.enabled && var.enable_cluster_autoscaler) ? 1 : 0
+  name   = "AmazonEKSNodeGroupAutoscaler"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  name   = "${module.label.id}-autoscaler"
```

let's use the label module to generate (unique and consistent) names for all resources.

Comment (aknysh) [reply main.tf]: please add the new policy attachment (`resource "aws_iam_role_policy_attachment" "amazon_eks_worker_node_autoscaler_policy"`) to the list here
https://github.com/cloudposse/terraform-aws-eks-node-group/blob/master/main.tf#L103

**Comment by maxgio92** [comment]
Hi @aknysh,
thank you! I'll fix all my code as you pointed out :)

**Comment by maxgio92** [comment]
Hi @aknysh,
I just pushed the recommended changes.
Thanks

**Comment by aknysh** [comment]
thanks @maxgio92 

can you also run

```
make init
make github/init
```

We need to push a few more files into the repo, and this PR is a good chance to do it.

Thanks

**Comment by maxgio92** [comment]
Hi @aknysh,
just pushed.

**Comment by maxgio92** [comment]
Thank you @aknysh!

**Pull request merged**